### ISSUE TITLE
.skip(0) is slow

### ISSUE THREAD

**Issue by Zolmeister** [rethinkdb/rethinkdb#4741]
Running a query with `skip(0)` requires 5x I/O operations (as reported by dashboard, compared to omitting the skip)

```js
r.table('table').skip(0).limit(10) // slow
r.table('table').limit(10) // fast
```

Edit: rethinkdb v2.0.4

**Comment by danielmewes** [comment]
I'm not entirely sure why this happens. @mlucy do you have an explanation for this?

**Comment by mlucy** [comment]
When you write `table.limit(10)`, RethinkDB picks a batch size for the shards that tries to load as few documents as possible from each shard while giving us a very high chance of successfully loading 10 documents in one batch if the shards are roughly balanced.  I would guess that `table.skip(0).limit(10)` the `skip` is discarding this batch optimization for some reason and using its own batch configuration, which causes all the shards to send back more data (some of which is then discarded because we only need 10 elements).

Probably writing `table.slice(a, a+b)` instead of `table.skip(a).limit(b)` would work as a workaround until we fix the batch sizing.

**Comment by mlucy** [comment]
Fixed in #5155.
